### ISSUE TITLE
Question about RAM usage

### ISSUE THREAD

**Issue by Liiw** [ValkyrienWarfare/Valkyrien-Warfare-Revamped#66]
Hello!

I was recently testing this mod along with numerous other mods on a SP world. I found that while I had 6G of RAM allocated to Java, the test ships I created, stopped functioning or "froze" in midair some time after they were subjected to physics- I couldn't interact with the vessels, they just sat there in midair or upside-down on the ground. I also noticed that when I increased the RAM allocation to 8G, the test vessel I created only froze for a short period of time. I have not tried the mod without any other mods, but I have a question on how much RAM would you think is appropriate for this mod to function normally? 

I might add, that I did have a Sphax x64 resourcepack installed if that somehow influences things. I am yet to test the mod on our Spongeforge server.

**Comment by BigBastard** [comment]
I believe the freezing occurred because of a dumb mistake I made with the render code. The issue was that sometimes Ships were loading faster than the game expected them to, and that resulted in the Ships expanding out their rendering code WAY beyond what they needed to. It must've been declaring thousands of unnecessary OpenGL displaylists. 

For servers though, as they don't do any rendering, this shouldn't be an issue.

Anyways I think I found the root cause of it, and I published a build that hopefully fixes it. Can you tell me how #94 works out? https://jenkins.ironhide.de/job/Valkyrien-Warfare-Revamped/94/

**Comment by BigBastard** [comment]
Edit: I found some new issues that also could've caused this. Try build 95 instead
https://jenkins.ironhide.de/job/Valkyrien-Warfare-Revamped/95/

**Comment by Liiw** [comment]
Hello again!

I tried build 96 and though the test ships were a bit more responsive, in the end they did not respond to commands from the hovercraft controller or the engines I placed onboard. I tried it two ways with the engines. 

First I tried to install engines on the craft AFTER infusing it with physics (using the creative infuser in both cases). And a lever next to the engines for controlling. The ship did not budge. I was able to break the ship and place blocks on the ship, but that was all. Functionally it was dead.

Secondly I installed the engine and lever BEFORE infusing the craft with physics. I also turned the lever on before activating the infuser. This ship immediately started moving towards where the engine was pushing it. I managed to raise the altitude using the hovercraft controller. After a brief flight of several seconds, the ship decided to activate it's cloaking device and jumped into hyperspace (i.e. it disappeared). I haven't heard from it since...

After that I built another ship which ultimately froze again. Standing on that ship, I initiated a manual crash with F3+C, hoping that the crash log could help somehow:

[crash-2017-03-08_04.30.23-client.txt](https://github.com/ValkyrienWarfare/Valkyrien-Warfare-Revamped/files/826462/crash-2017-03-08_04.30.23-client.txt)

And the fml log:

[fml-client-latest.txt](https://github.com/ValkyrienWarfare/Valkyrien-Warfare-Revamped/files/826466/fml-client-latest.txt)

**Comment by DaMatrix** [comment]
You might have accidentally paused physics on the ship, have you tried right-clicking the physics infuser again?

**Comment by Liiw** [comment]
Hello again!

Unfortunately that was not the case. I loaded up a fresh world and built some hoverbarges to test this. I raised the altitude to around 100 and waited a little, making sure I would not touch the infuser. After that I wanted to decrease the altitude, but it would not after that. I also had an incident before that, in which, upon infusing the physics, the barge fell on the ground, bounced on it's side and then froze like that.

On a bit unrelated note. Are RF-powered hoverengines and regular engines to be implemented in the future?
